src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -1716,35 +1729,34 @@ namespace rascal {
                                                                      neigh);
         double f_c{cutoff_function->f_c(dist)};
         auto coefficients_center_by_type{coefficients_center[neigh_type]};
-
-        // compute the coefficients
         size_t l_block_idx{0};
-        for (size_t angular_l{0}; angular_l < this->max_angular + 1;
-             ++angular_l) {
-          size_t l_block_size{2 * angular_l + 1};
-          c_ij_nlm.block(0, l_block_idx, max_radial, l_block_size) =
-              neighbour_contribution.col(angular_l) *
-              harmonics.segment(l_block_idx, l_block_size);
-          l_block_idx += l_block_size;
-        }
-        c_ij_nlm *= f_c;
-        coefficients_center_by_type += c_ij_nlm;
-
-        // half list branch for c^{ji} terms using
-        // c^{ij}_{nlm} = (-1)^l c^{ji}_{nlm}.
-        if (IsHalfNL) {
-          if (not manager->is_center_atom(atom_j)) {
-            std::stringstream err_str{};
-            err_str << "Half neighbor list should only be used when all the "
-                    << "atoms inside the unit cell are centers, i.e. "
-                    << "center_atoms_mask should not mask atoms.";
-            throw std::runtime_error(err_str.str());
+        // compute the coefficients
+        // complications here because the infrastructure for swaping ij is not
+        // here yet.
+        // TODO(felix) make swaping ij possible and implement the
+        // proper half for gradients so the whole computation can be skipped.
+        if (not are_some_centers_masked and atom_i_tag > atom_j_tag) {
+          // continue;
+        } else {
```

Comment:
Is the `// continue` a placeholder for anything you want to implement later? Otherwise replacing the if-else with just `if (are_any_centers_masked or (atom_i_tag <= atom_j_tag)) {` (the negation of the current statement) would be clearer.

Also, please use more parentheses around the negation and the comparison; it helps readability and avoids confusion due to having to remember C++'s order of operations.

Reply:
when the swap ij functionality is there (which is soon) there will be a 
```
        if (not are_some_centers_masked and atom_i_tag > atom_j_tag) {
          continue;
        }
```
at the very top of the iteration over the pairs so writing it like this make the future change simpler.

Reply:
Ok. And I really do prefer putting parentheses around the operands of a boolean operator, both because it avoids anyone reading this as `if not (are any centers masked and atom_i_tag > atom_j_tag)` (that's how I first read this) and having to remember that both `not` and `>` are higher in this table: https://en.cppreference.com/w/cpp/language/operator_precedence than the `and` operator.

From the coding convention, section https://cosmo-epfl.github.io/librascal/dev_guide/coding-convention.html#boolean-expressions:

> Feel free to insert extra parentheses judiciously because they can be very helpful in increasing readability when used appropriately.